pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -243,25 +259,34 @@ func (m *MachineScope) IsStackHub() bool {
 	return strings.EqualFold(m.cloudEnv, string(configv1.AzureStackCloud))
 }
 
-func GetCloudEnvironment(client controllerclient.Client) (string, string, error) {
+func GetCloudEnvironment(client controllerclient.Client) (map[string]string, string, string, string, error) {
 	infra := &configv1.Infrastructure{}
 	infraName := controllerclient.ObjectKey{Name: globalInfrastuctureName}
+	tags := make(map[string]string)
 
 	if err := client.Get(context.Background(), infraName, infra); err != nil {
-		return "", "", err
+		return tags, "", "", "", err
+	}
+
+	fmt.Printf("----------------------\n%+v\n-------------------------\n", infra)
```

Comment:
Debug logic?

Reply:
Yeah, will remove the commit, added to debug unit workflow failure.